dev: update autocommit_selectforupdate_nowaitlock.png (#22108)
pingcap/docs: develop/dev-guide-transaction-restraints.md

## Auto-committed `SELECT FOR UPDATE` statements do NOT wait for locks

> Currently, locks are not added to auto-committed `SELECT FOR UPDATE` statements. The effect is shown in the following screenshot. The left and right parts of the screenshot are two different sessions.

Author: ```suggestion
Currently, locks are not added to auto-committed `SELECT FOR UPDATE` statements. The following screenshot shows the effect in two separate sessions:
```
